ui/src/file_upload.tsx

```diff
@@ -113,29 +113,42 @@ export const
       maxFileSizeBytes = max_file_size ? convertMegabytesToBytes(max_file_size) : 0,
       maxSizeBytes = max_size ? convertMegabytesToBytes(max_size) : 0,
       fileExtensions = file_extensions ? file_extensions.map(e => e.startsWith('.') ? e : `.${e}`) : null,
+      isAborted = React.useRef<B>(false),
```

Comment:
No need for extra ref to check if the request was aborted or not. See [this answer](https://stackoverflow.com/questions/3648309/how-to-detect-if-a-request-was-aborted), the `status` should equal `0` in case of abort.
```suggestion
```

Reply:
We could not relay on `0` value of the `status` since it equals zero also on other circumstances than request aborted by user (e.g. lost of internet connection). Also we can't relay on `responseText` and `statusText` which are empty values in our case.

![image](https://user-images.githubusercontent.com/23740173/172176727-24f77840-501a-4e73-a20e-440ede45bf4f.png)